docs/src/pages/docs/crocks/Result.md

```diff
@@ -0,0 +1,1422 @@
+  chain(ensure(isDefined)),
+  tryCatch(prop('age'))
+)
+
+tryGetAge({ name: 'Sarah', age: 23 })
+//=> Ok 23
+
+tryGetAge({ name: 'Sarah' })
+//=> Err "Age not found!"
+
+tryGetAge(1)
+//=> Err "Age not found!"
+
+tryGetAge(undefined)
+//=> Err "Age not found!"
+```
+
+<article id="topic-implements">
+
+## Implements
+`Setoid`, `Semigroup`, `Functor`, `Alt`, `Apply`, `Traversable`,
+`Chain`, `Applicative`, `Monad`
+
+</article>
+
+<article id="topic-construction">
+
+## Construction
+
+```haskell
+Result :: a -> Result e a
+```
+
+Most of the time, `Result` is constructed using functions of your own making
+and helper functions like [`tryCatch`](#trycatch) or by employing one of the
+instance constructors, [`Ok`](#ok) or [`Err`](#err). This is due to the nature
+of `Result` and most other Sum Types.
+
+As a matter of consistency and completion, a `Result` instance can also be
+constructed using its TypeRep like any other type. The `Result` constructor is
+a unary function that accepts any type `a` and returns a [`Ok`](#ok) instance,
+wrapping the value passed to its argument.
+
+```javascript
+import Result from 'crocks/Result'
+
+import equals from 'crocks/pointfree/equals'
+
+const { Ok, Err, of } = Result
+
+Result('some string')
+//=> Ok "some string"
+
+Result(null)
+//=> Ok null
+
+Result(undefined)
+//=> Ok undefined
+
+of('some string')
+//=> Ok "some string"
+
+of(null)
+//=> Ok null
+
+of(undefined)
+//=> Ok undefined
+
+Ok('some string')
+//=> Ok "some string"
+
+Ok(null)
+//=> Ok null
+
+Ok(undefined)
+//=> Ok undefined
+//=> Ok undefined
+
+Err('some string')
+//=> Err "some string"
+
+Err(null)
+//=> Err null
+
+Err(undefined)
+//=> Err undefined
+
+equals(
+  Result.Ok([ 1, 2, 3 ]),
+  Result.of([ 1, 2, 3 ])
+)
+//=> true
+
+equals(
+  of({ a: 100 }),
+  Result({ a: 100 })
+)
+//=> true
+```
+
+</article>
+
+<article id="topic-constructor">
+
+## Constructor Methods
+
+#### Err
+
+```haskell
+Result.Err :: e -> Result e a
+```
+
+Used to construct an [`Err`](#err) instance that represents the "false" or
+"Negative" portion of a disjunction. When an instance is an [`Err`](#err), most
+`Result` returning methods will just return a new [`Err`](#err) instance with
+the same containing value.
+
+The power of the [`Err`](#err) as opposed to a [`Nothing`] is that it can hold
+meaningful information on why the flow is in this path. This works as a core
+tool when using Railway Orientated Programming concepts.
+
+```javascript
+import Result from 'crocks/Result'
+
+import chain from 'crocks/pointfree/chain'
+import bimap from 'crocks/pointfree/bimap'
+import isNumber from 'crocks/predicates/isNumber'
+import ifElse from 'crocks/logic/ifElse'
+import compose from 'crocks/core/compose'
+
+const { Ok, Err } = Result
+
+// ensure :: (a -> bool) -> (a -> Result a a)
+const ensure = pred => ifElse(pred, Ok, Err)
+
+// buildError :: String -> String
+const buildError = x => `${x} is not a valid number`
```

Comment:
While it is not too bad, we probably wanna shy away from saying it is okay to put values in error messages. Imagine if we were checking a password or pulling in critical or private data, this would allow for private data to trickle into logs that may or may not be scrubbed or secured. May wanna change what this error reports. Also should be in two line format:

```js
// buildError :: () -> String
const buildError = () =>
  'number not provided'
```

Reply:
Yeah i get where you're coming from here for sure. This was more to show the difference from a `Nothing` in that it held a value. Would it be better to just change the example to something different that shows that? or are we happy to show an example without the value?

Reply:
I think it should be fine without the value.